src/AggregateFunctions/AggregateFunctionWelchTTest.cpp

```diff
@@ -52,6 +52,14 @@ struct WelchTTestData : public TTestMoments<Float64>
         Float64 se = getStandardError();
         Float64 t_stat = (mean_x - mean_y) / se;
 
+        auto degrees_of_freedom = getDegreesOfFreedom();
+
+        if (degrees_of_freedom <= 0 || std::isinf(degrees_of_freedom) || isNaN(degrees_of_freedom))
```

Comment:
maybe better to not repeat this logic (ctor already checks its arguments), but just catch `domain_error` from ctor?

Reply:
I mean `students_t_distribution`'s constructor, check input arguments is its responsibility and according to the [docs](https://www.boost.org/doc/libs/1_80_0/libs/math/doc/html/math_toolkit/dist_ref/dists/students_t_dist.html#math_toolkit.dist_ref.dists.students_t_dist.member_functions) it has some checks